README.adoc

```diff
@@ -112,47 +128,116 @@ NOTE: The `+"auth":"<username>:<password>"+` field must be base64-encoded.
+|===
 
-These artifact IDs values can be accessed in subsequent steps using the outputs context. For example, to return:
+=== Using the action output
 
-* All artifact ID values provided by action in JSON format use: `${{ steps.<action_step_name>.outputs.artifactIds }}`.
+Access the `artifactIds` values in downstream steps using the `outputs` xref:dsl-syntax:contexts.adoc[context].
+To use the following examples, replace <ACTION_STEP_NAME> with the action step name, and <DESTINATION_URL> with the destination URL.
 
-* A specific artifact ID value use: `${{ fromJSON(steps.<action_step_name>.outputs.artifactIds).<destination URL>}}`.
+* For all artifact ID values in JSON format provided by the action, use: `${{ steps.<ACTION_STEP_NAME>.outputs.artifactIds }}`.
 
-For instance kaniko action was used in the step with name "kaniko-build", and the destination URL is `registry.example.com/my-image:latest`, then you can access the artifact ID using:
-`${{ fromJSON(steps.kaniko-build.outputs.artifactIds)['registry.example.com/my-image:latest'] }}`.
+* For a specific artifact ID value use: `${{ fromJSON(steps.<ACTION_STEP_NAME>.outputs.artifactIds).<DESTINATION_URL>}}`.
 
-|===
+== Usage examples
 
-== Usage example
+=== Basic example
 
-In your YAML file, add:
+The following is a basic usage example for this action:
 
 [source,yaml]
 ----
-      - id: kaniko-build
-        name: Build a container image with Kaniko
+      - name: Build a container image with Kaniko
         uses: cloudbees-io/kaniko@v1
         with:
-          dockerfile: path/to/Dockerfile
-          context: .
-          destination: <registry host>/<image name>:${{cloudbees.version}},<registry host>/<image name>:latest
+          destination: path/to/registry/host/my-image:1.0.1,path/to/registry/host/my-image:latest
+----
+
+=== Using optional inputs
+
+The following example specifies optional inputs:
+
+[source,yaml]
+----
+      - name: Kaniko build with optional inputs
+        uses: cloudbees-io/kaniko@v1
+        with:
+          destination: path/to/registry/host/my-image:1.0.1,path/to/registry/host/my-image:latest
           build-args: BUILDKIT_CONTEXT_KEEP_GIT_DIR=1,BUILDKIT_INLINE_CACHE=1
-          labels: maintainer=John Smith,version=${{cloudbees.version}}
-          verbosity: info
-          tar-path: path/to/save/image.tar
+          context: .
+          dockerfile: path/to/Dockerfile
+          labels: maintainer=John Smith,version=1.0.1
+          tar-path: path/to/image.tar
+          verbosity: warn
+
+----
 
-      - id: echo-artifact-ids
+=== Full workflow example
+
+The following workflow example:
+
+* Checks out source code from a repository.
+* Configures Docker credentials.
+* Builds and publishes a container image with Kaniko.
+* Prints the artifact IDs for both static and dynamically created destinations.
+* Promotes the image to AWS ECR.
+* Accesses the `artifactIds` value in the step to promote the image to ECR.
```

Comment:
The promotion step does not use the artifactIds so we should remove it from here. In future, when we enhance the `publish-deployed-artifact` action to accept artifactId as input, we could expand the example then. For now, lets remove it.
And so also, remove the step "Promote an image in ECR" from the workflow DSL below.